### ISSUE TITLE
Extract intermediates into a list instead of printing?

### ISSUE THREAD

**Issue by nischalshrestha** [moodymudskipper/boomer#72]
Hello! I would like to just say package is very cool and closely aligned with the vision I had for [Unravel](https://github.com/nischalshrestha/Unravel)---to provide easy access/exploration of intermediates for tidyverse code. I would love to one day do it for any call (base or tidyverse) as you have done.

Let me know if I've missed it, but is there any way to gain access to list of intermediates instead of printing? I had this same dilemma with Unravel and decided to have a public function that would give you the list via `get_chain_outputs()`.

**Comment by moodymudskipper** [comment]
Hi Nishal and thanks.
One reason why boom doesn't return the printed data is that a common use case will be that the call fails, and we still want to print something.
The calls are printed as the code is executed, which is nice for long operations too, no need to wait to see output.
However your request makes sense and I believe I can implement something close enough by feeding a structure stored in a global variable as the code is executed. We could then access it with something like `last_boom()`.

**Comment by nischalshrestha** [comment]
Makes sense! But, I do think that if a call fails in the steps that's also informative, it's just tricky how you want that to be represented. I faced the same type of challenge before. Something I've seen recently is [purrr:safely](https://purrr.tidyverse.org/reference/safely.html) but that is an extra dependency which isn't needed. A similar concept like that would be cool though.

**Comment by moodymudskipper** [comment]
For 1.0 an experimental function `boom_data()` gathers the data in a data frame, we don't go further than this as a start. later we can have args to `boom_data()` to do more.
